File: src/main/java/io/github/dqualizer/dqlang/types/dam/domainstory/ValueObject.java
package io.github.dqualizer.dqlang.types.dam.domainstory;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.annotation.PersistenceCreator;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
@Getter
public class ValueObject extends DSTElement {
    @JsonCreator
    @PersistenceCreator
    public ValueObject(@JsonProperty("name") @NotNull String name) {
        super(name);
    }
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/dam/domainstory/ValueObject.kt
package io.github.dqualizer.dqlang.types.dam.domainstory

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
data class ValueObject @JsonCreator constructor(
    override val name: String
): DSTElement(name)

